src/assets/wise5/directives/componentAnnotations/component-annotations.component.ts

```diff
@@ -45,34 +57,88 @@ export class ComponentAnnotationsComponent {
         }
       }
     );
+    this.wiseLinkCommunicatorId = `wise-link-communicator-component-annotations-${this.nodeId}-${this.componentId}`;
   }
 
-  ngOnChanges() {
+  ngAfterViewInit() {
     this.processAnnotations();
+    this.wiseLinkCommunicator = document.getElementById(this.wiseLinkCommunicatorId);
+    if (this.wiseLinkCommunicator != null) {
+      this.wiseLinkClickedHandler = this.WiseLinkService.createWiseLinkClickedHandler(this.nodeId);
+      this.WiseLinkService.addWiseLinkClickedListener(
+        this.wiseLinkCommunicator,
+        this.wiseLinkClickedHandler
+      );
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (!changes.annotations.isFirstChange()) {
+      this.processAnnotations();
+    }
   }
 
   ngOnDestroy() {
     this.studentWorkSavedToServerSubscription.unsubscribe();
+    if (this.wiseLinkClickedHandler != null) {
```

Comment:
Is this null check necessary?

Reply:
The null check is there for the same reason above.